test/cases/enum.ts

```diff
@@ -6,7 +6,8 @@ export var schema = {
       "enum": ["a", "b", "c"]
     },
     "bar": {
-      "enum": [1, 2, 3]
+      "enum": [1, 2, 3],
+      "tsEnumNames": ["One","Two","Three"]
```

Comment:
So what you're saying is

1. If `enum` is a list of strings, then compile to a TS enum
2. If `enum` is a list of strings and the key `enumValues` exists and is a list of numbers, use `enumValues` as the values for the keys defined in `enum`
3. If `enum` is a list of numbers and the key `enumNames` exists and is a list of strings, use `enum` as the values for the  keys defined in `enumNames`
4. If (a) `enum` is not a list of strings, and (b) `enum` is not a list of numbers and `enumNames` is not defined, throw a TypeError
5. If `enumValues` is defined and is not a list of numbers, throw a TypeError
6.  If `enumNames` is defined and is not a list of strings, throw a TypeError

SGTM.